pkg/didcomm/protocol/introduce/service.go

```diff
@@ -108,7 +112,7 @@ type Provider interface {
 // Forwarder provides the possibility to forward an invitation
 type Forwarder interface {
 	// method should be implemented in didexchange service
-	SendInvitation(inv *didexchange.Invitation, dest *service.Destination) error
+	SendInvitation(pthID string, inv *didexchange.Invitation, dest *service.Destination) error
```

Comment:
Are you planning to create a new Invitation object based on `inv` and `pthID`?

Reply:
To close the state machine I need the same thread id. According to that, I assume that did-exchange will start protocol (did-exchange) with parent thread id.